### PR TITLE
Adding example Wait block to interpretter demo.

### PR DESCRIPTION
**Pull request by AnmAtAnm**
This change is [Reviewable](https://reviewable.io/reviews/google/blockly/1001)

### CODE REVIEW

File: demos/interpreter/wait_block.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+Blockly.JavaScript['wait_seconds'] = function(block) {
+  var seconds = Number(block.getFieldValue('SECONDS'));
+  var code = 'waitForSeconds(' + seconds + ');\n';
+  return code;
+};
+
+/**
+ * Register the interpreter asynchronous function <code>waitForSeconds()</code>.
+ */
+function initInterpretterWaitForSeconds(interpreter, scope) {
```
Comment (rachel-fenichel) [inline demos/interpreter/wait_block.js]: Interpreter is misspelled (spelled interpretter here).  Please fix.

File: demos/interpreter/index.html [hunk 2]
```diff
@@ -167,6 +174,7 @@
       Blockly.JavaScript.STATEMENT_PREFIX = 'highlightBlock(%1);\n';
       Blockly.JavaScript.addReservedWords('highlightBlock');
       var code = Blockly.JavaScript.workspaceToCode(workspace);
+      console.log(code);
```
Comment (rachel-fenichel) [inline demos/interpreter/index.html]: Delete this line.

File: demos/interpreter/wait_block.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
+/**
+ * @license
+ * Visual Blocks Editor
+ *
+ * Copyright 2012 Google Inc.
```
Comment (rachel-fenichel) [inline demos/interpreter/wait_block.js]: Fix year

File: demos/interpreter/wait_block.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Example "wait" block that will pause the interpretter for a
```
Comment (rachel-fenichel) [inline demos/interpreter/wait_block.js]: Fix spelling of "interpreter"

File: demos/interpreter/wait_block.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+/**
+ * Register the interpreter asynchronous function <code>waitForSeconds()</code>.
+ */
+function initInterpreterWaitForSeconds(interpreter, scope) {
+  var wrapper = interpreter.createAsyncFunction(function(timeInSeconds, callback) {
+    // Delay the call to the callback.
+    setTimeout(callback, timeInSeconds * 1000);
+  });
+  interpreter.setProperty(scope, "waitForSeconds", wrapper);
```
Comment (NeilFraser) [inline demos/interpreter/wait_block.js]: Prefer single quotes.

**Comment by AnmAtAnm** [comment]
I think I'm going to refactor this to separate the wait block into another demo.  Currently, it crashes during stepping (stack overflow on recursion). To solve this will require the wait block to have too much integration with the `highlightPause` variable than a clear/useful wait demo should have.

**Comment by AnmAtAnm** [comment]
PTAL
 * Factored out the async components for clarity, so now we have two interpreter demos.
 * Cleaned up the UI, removing most of the modal alerts. Program output is now in a `<textarea>`.
 * Both demo now automatically load the block code into the interpreter, eliminating the "Parse JavaScript" button.

**Comment by RoboErikG** [comment]
:lgtm: after a few nits, but wait for Rachel's pass.

---

Review status: 0 of 5 files reviewed at latest revision, 6 unresolved discussions.

---

*[demos/interpreter/step-execution.html, line 32 at r6](https://reviewable.io:443/reviews/google/blockly/1001#-Kj5CmQsylulv9Nt9GNg:-Kj5CmQsylulv9Nt9GNh:b-4dwye3) ([raw file](https://github.com/google/blockly/blob/a12f750c17bdb0a67b3119c5940e2f7c87f09d47/demos/interpreter/step-execution.html#L32)):*
> ```HTML
>   and set the variable <code>highlightPause</code> to <code>true</code>.</p>
> 
>   <p>"Parse JavaScript" will generate the code and load it into the interpreter. Then, aach press of the
> ```

s/aach/each/

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly/1001#-:-Kj5D2_JCfehTXx03NCY:bg1zr0f)*

**Comment by rachel-fenichel** [comment]
Reviewed 1 of 4 files at r6.
Review status: 1 of 5 files reviewed at latest revision, 8 unresolved discussions.

---

*[demos/interpreter/async-execution.html, line 162 at r6](https://reviewable.io:443/reviews/google/blockly/1001#-Kj5F8ctkp7PiqOXZ7sl:-Kj5F8ctkp7PiqOXZ7sm:be05b3d) ([raw file](https://github.com/google/blockly/blob/a12f750c17bdb0a67b3119c5940e2f7c87f09d47/demos/interpreter/async-execution.html#L162)):*
> ```HTML
>         return interpreter.createPrimitive(prompt(text));
>       };
>       interpreter.setProperty(scope, 'prompt',
> ```

You're using two different patterns for creating the native function and setting it as a property.  Please choose one.  I lean toward this second one because it's more explicit.

---

*[demos/interpreter/index.html, line 177 at r1](https://reviewable.io:443/reviews/google/blockly/1001#-Kh44MbL4PjxUfycGHYE-r1-177:-Kj5Fl1hA2JppDzifSjr:batz1b5) ([raw file](https://github.com/google/blockly/blob/700e4fe0bfd0d0b5b2b5cf32152f1342b978023f/demos/interpreter/index.html#L177)):*
<details><summary><i>Previously, rachel-fenichel (Rachel Fenichel) wrote…</i></summary><blockquote>

Delete this line.
</blockquote></details>

My comment seems thoroughly obsolete now.

---

*[demos/interpreter/step-execution.html, line 87 at r6](https://reviewable.io:443/reviews/google/blockly/1001#-Kj5GE3BDXs_8_Kj6Z7n:-Kj5GE3BDXs_8_Kj6Z7o:biyvwqz) ([raw file](https://github.com/google/blockly/blob/a12f750c17bdb0a67b3119c5940e2f7c87f09d47/demos/interpreter/step-execution.html#L87)):*
> ```HTML
>   </xml>
> 
>   <xml id="startBlocks" style="display: none">
> ```

Here and in the other demo: Since you're displaying the block IDs to the user, it makes sense to set them explicitly to reasonable values instead of letting them be long UUIDs.

---

*[demos/interpreter/wait_block.js, line 24 at r6](https://reviewable.io:443/reviews/google/blockly/1001#-Kj5G6GLKutUhSYRorRt:-Kj5G6GLKutUhSYRorRu:b-7qiorf) ([raw file](https://github.com/google/blockly/blob/a12f750c17bdb0a67b3119c5940e2f7c87f09d47/demos/interpreter/wait_block.js#L24)):*
> ```JavaScript
>  * @fileoverview Example "wait" block that will pause the interpreter for a
>  * number of seconds. Because wait is a blocking behavior, such blocks will
>  * only work in interpretted environments.
> ```

Fix spelling of "interpreted"

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly/1001)*

**Comment by AnmAtAnm** [comment]
Review status: 1 of 5 files reviewed at latest revision, 9 unresolved discussions.

---

*[demos/interpreter/async-execution.html, line 162 at r6](https://reviewable.io:443/reviews/google/blockly/1001#-Kj5F8ctkp7PiqOXZ7sl:-Kj5RvxCm1oQ-OY4oj9C:b-896fix) ([raw file](https://github.com/google/blockly/blob/a12f750c17bdb0a67b3119c5940e2f7c87f09d47/demos/interpreter/async-execution.html#L162)):*
<details><summary><i>Previously, rachel-fenichel (Rachel Fenichel) wrote…</i></summary><blockquote>

You're using two different patterns for creating the native function and setting it as a property.  Please choose one.  I lean toward this second one because it's more explicit.
</blockquote></details>

Done.

---

*[demos/interpreter/step-execution.html, line 32 at r6](https://reviewable.io:443/reviews/google/blockly/1001#-Kj5CmQsylulv9Nt9GNg:-Kj5OgBmJctXA7gcyAiH:b-896fix) ([raw file](https://github.com/google/blockly/blob/a12f750c17bdb0a67b3119c5940e2f7c87f09d47/demos/interpreter/step-execution.html#L32)):*
<details><summary><i>Previously, RoboErikG wrote…</i></summary><blockquote>

s/aach/each/
</blockquote></details>

Done.

---

*[demos/interpreter/step-execution.html, line 87 at r6](https://reviewable.io:443/reviews/google/blockly/1001#-Kj5GE3BDXs_8_Kj6Z7n:-Kj5QNqjseqXgFif0hE2:b-p0ktnd) ([raw file](https://github.com/google/blockly/blob/a12f750c17bdb0a67b3119c5940e2f7c87f09d47/demos/interpreter/step-execution.html#L87)):*
<details><summary><i>Previously, rachel-fenichel (Rachel Fenichel) wrote…</i></summary><blockquote>

Here and in the other demo: Since you're displaying the block IDs to the user, it makes sense to set them explicitly to reasonable values instead of letting them be long UUIDs.
</blockquote></details>

Done for statement blocks.

---

*[demos/interpreter/wait_block.js, line 5 at r1](https://reviewable.io:443/reviews/google/blockly/1001#-Kh44MbL4PjxUfycGHYF-r1-5:-Kj5QZq-d4IuIIkVkUEW:b-896fix) ([raw file](https://github.com/google/blockly/blob/700e4fe0bfd0d0b5b2b5cf32152f1342b978023f/demos/interpreter/wait_block.js#L5)):*
<details><summary><i>Previously, rachel-fenichel (Rachel Fenichel) wrote…</i></summary><blockquote>

Fix year
</blockquote></details>

Done.

---

*[demos/interpreter/wait_block.js, line 22 at r1](https://reviewable.io:443/reviews/google/blockly/1001#-Kh44MbL4PjxUfycGHYF-r1-22:-Kj5Qbfim5CMJVPadxgp:b-896fix) ([raw file](https://github.com/google/blockly/blob/700e4fe0bfd0d0b5b2b5cf32152f1342b978023f/demos/interpreter/wait_block.js#L22)):*
<details><summary><i>Previously, rachel-fenichel (Rachel Fenichel) wrote…</i></summary><blockquote>

Fix spelling of "interpreter"
</blockquote></details>

Done.

---

*[demos/interpreter/wait_block.js, line 52 at r1](https://reviewable.io:443/reviews/google/blockly/1001#-Kh44MbL4PjxUfycGHYF-r1-52:-Kj5S8eKUvs3yp3IhaMg:b-896fix) ([raw file](https://github.com/google/blockly/blob/700e4fe0bfd0d0b5b2b5cf32152f1342b978023f/demos/interpreter/wait_block.js#L52)):*
<details><summary><i>Previously, rachel-fenichel (Rachel Fenichel) wrote…</i></summary><blockquote>

Interpreter is misspelled (spelled interpretter here).  Please fix.
</blockquote></details>

Done.

---

*[demos/interpreter/wait_block.js, line 57 at r2](https://reviewable.io:443/reviews/google/blockly/1001#-Kh44MbL4PjxUfycGHYF-r2-57:-Kj5S9tA509RPNsK4z88:b-896fix) ([raw file](https://github.com/google/blockly/blob/1b5a2f81ccc34c7b264bc2bbd33a6dd00679476b/demos/interpreter/wait_block.js#L57)):*
<details><summary><i>Previously, NeilFraser (Neil Fraser) wrote…</i></summary><blockquote>

Prefer single quotes.
</blockquote></details>

Done.

---

*[demos/interpreter/wait_block.js, line 24 at r6](https://reviewable.io:443/reviews/google/blockly/1001#-Kj5G6GLKutUhSYRorRt:-Kj5S3JxjYOl0xGm6ilp:b-896fix) ([raw file](https://github.com/google/blockly/blob/a12f750c17bdb0a67b3119c5940e2f7c87f09d47/demos/interpreter/wait_block.js#L24)):*
<details><summary><i>Previously, rachel-fenichel (Rachel Fenichel) wrote…</i></summary><blockquote>

Fix spelling of "interpreted"
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly/1001)*

**Comment by rachel-fenichel** [comment]
lgtm

**Pull request merged**